fri/test.py

```diff
@@ -23,37 +23,37 @@ def build(dir, graphml, apikey):
   print(response.text)
 
 # function to debug
-def debug(graphml):
-  url = "http://127.0.0.1:5000/debug/"+graphml
+def debug(graphml, apikey):
+  url = "http://127.0.0.1:5000/debug/"+graphml+"?"+"apikey="+apikey
   response = requests.request("POST", url)
   print(response.text) 
 
 # function to test run() method.
-def run(graphml):
-  url = "http://127.0.0.1:5000/run/"+graphml
+def run(graphml, apikey):
+  url = "http://127.0.0.1:5000/run/"+graphml+"?"+"apikey="+apikey
   response = requests.request("POST", url)
   print(response.text)
 
-def clear(graphml):
-  url = "http://127.0.0.1:5000/clear/"+graphml
+def clear(graphml, apikey):
+  url = "http://127.0.0.1:5000/clear/"+graphml+"?"+"apikey="+apikey
   response = requests.request("POST", url)
   print(response.text)
 
-def stop(graphml):
-  url = "http://127.0.0.1:5000/stop/"+graphml
+def stop(graphml, apikey):
+  url = "http://127.0.0.1:5000/stop/"+graphml+"?"+"apikey="+apikey
   response = requests.request("POST", url)
   print(response.text)    
 
 
 #function to destroy dir.
-def destroy(dir):
-  url = "http://127.0.0.1:5000/destroy/" + dir
+def destroy(dir, apikey):
+  url = "http://127.0.0.1:5000/destroy/" + dir+"?"+"apikey="+apikey
   response = requests.request("DELETE", url)
 
   print(response.text)  
   
-def getFilesList(dir, sub_dir = ""):
-  url = "http://127.0.0.1:5000/getFilesList/" + dir + "?"+"fetch="+sub_dir
+def getFilesList(apikey, dir, sub_dir = ""):

```

Comment:
In your other methods, apikey was the last argument. Please keep the consistency and leave it as the last argument.

Reply:
> In your other methods, apikey was the last argument. Please keep the consistency and leave it as the last argument.

Sure, I will change this.

> Thanks, @Rahuljagwani.
> 
> I have added a few points. But the most important of the comments is any changes to this should be limited to only FRI (both main.py and test.py.) Please do not include any changes to mkconcre.py in this PR.
> 
> Please let me know if I am missing something or if something is unclear.
> 
> Even for the GSoC, I mostly see the changes in concore-editor and FRI only. FRI functions as the REST server backend for the concore-editor, creating a bridge between concore and concore-editor.

@pradeeban First of all, thanks for reviewing my code .

**Changes proposed for this PR**
1. Firstly, all the changes from mkconcore.py must be reverted.
2. A check must be added in main.py for apikeys. For now, whether or not user is passing them it is completely fine. But for future perspects of Kong they must be passed therefore this check will become optional. When apikeys are not passed no appending of directories with "_", otherwise process will be same.
3. No changes in test.py except the small change of getfiles function's argument passing.

Therefore, for now all functions inside main.py will take apikey argument but not do anything significantly with it, except (build, uploadfiles)

I will be making above changes.
If there is anything I missed or any new thing proposed, please convey them here.
Thank You

Reply:
Yes, as long as you keep the changes to FRI and ensuring your follow-up changes do not break any of the progress that you have made in this integration, it is cool. We can always go incrementally.

But we should definitely *not* make "API Key" a mandatory argument. The rationale is not everyone will want to use the Kong (although if one uses our Dockerfile, it will eventually have the Kong -- unless we change plans later).

Is it possible to make it an optional argument?